library/data/registry-entry-attributes/src/main/java/org/quiltmc/qsl/registry/attribute/api/RegistryEntryAttribute.java

```diff
@@ -0,0 +1,289 @@
+	 *
+	 * @param registry registry this attribute is tied to
+	 * @param id identifier
+	 * @param defaultValue default value
+	 * @param <R> registry entry type
+	 * @return a new attribute
+	 */
+	static <R> RegistryEntryAttribute<R, Double> createDouble(Registry<R> registry, Identifier id, double defaultValue) {
+		return create(registry, id, Codec.DOUBLE, defaultValue);
+	}
+
+	/**
+	 * Creates a new {@code String} property.
+	 *
+	 * @param registry registry this attribute is tied to
+	 * @param id identifier
+	 * @param defaultValue default value
+	 * @param <R> registry entry type
+	 * @return a new attribute
+	 */
+	static <R> RegistryEntryAttribute<R, String> createString(Registry<R> registry, Identifier id, String defaultValue) {
+		return create(registry, id, Codec.STRING, defaultValue);
+	}
+
+	/**
+	 * Creates a new {@code boolean} property.
+	 *
+	 * @param registry registry this attribute is tied to
+	 * @param id identifier
+	 * @param <R> registry entry type
+	 * @return a new attribute
+	 */
+	static <R> RegistryEntryAttribute<R, Boolean> createBool(Registry<R> registry, Identifier id) {
+		return create(registry, id, Codec.BOOL, null);
+	}
+
+	/**
+	 * Creates a new {@code int} property.
+	 *
+	 * @param registry registry this attribute is tied to
+	 * @param id identifier
+	 * @param <R> registry entry type
+	 * @return a new attribute
+	 */
+	static <R> RegistryEntryAttribute<R, Integer> createInt(Registry<R> registry, Identifier id) {
+		return create(registry, id, Codec.INT, null);
+	}
+
+
+	/**
+	 * Creates a new {@code long} property.
+	 *
+	 * @param registry registry this attribute is tied to
+	 * @param id identifier
+	 * @param <R> registry entry type
+	 * @return a new attribute
+	 */
+	static <R> RegistryEntryAttribute<R, Long> createLong(Registry<R> registry, Identifier id) {
+		return create(registry, id, Codec.LONG, null);
+	}
+
+	/**
+	 * Creates a new {@code float} property.
+	 *
+	 * @param registry registry this attribute is tied to
+	 * @param id identifier
+	 * @param <R> registry entry type
+	 * @return a new attribute
+	 */
+	static <R> RegistryEntryAttribute<R, Float> createFloat(Registry<R> registry, Identifier id) {
+		return create(registry, id, Codec.FLOAT, null);
+	}
+
+	/**
+	 * Creates a new {@code double} property.
+	 *
+	 * @param registry registry this attribute is tied to
+	 * @param id identifier
+	 * @param <R> registry entry type
+	 * @return a new attribute
+	 */
+	static <R> RegistryEntryAttribute<R, Double> createDouble(Registry<R> registry, Identifier id) {
+		return create(registry, id, Codec.DOUBLE, null);
+	}
+
+	/**
+	 * Creates a new {@code String} property.
+	 *
+	 * @param registry registry this attribute is tied to
+	 * @param id identifier
+	 * @param <R> registry entry type
+	 * @return a new attribute
+	 */
+	static <R> RegistryEntryAttribute<R, String> createString(Registry<R> registry, Identifier id) {
+		return create(registry, id, Codec.STRING, null);
+	}
```

Comment:
I feel it would be more clear if any of these functions with default values or functional getters were named differently. This would make it more clear to a user of the API why the extra arguments exist.

Reply:
The functional getters are exclusive to the "dispatched" utility methods, though I could rename all the default value variants to `createXWithDefault`